### PR TITLE
Improve child process spawning

### PR DESCRIPTION
**Pull request by RaeesBhatti**
**- Summary**
If the path to binary is not quoted on Windows and contains a space, it will get the path wrong and fail. This change makes sure that the path to app binary is quoted.
Fixes: https://github.com/netlify/cli/issues/817

**- Test plan**
* Have your app binary in path with spaces
* Run `netlify dev`

**- Description for the changelog**
Quote app binary path before spawning it.

**- A picture of a cute animal (not mandatory but encouraged)**
🐬

### CODE REVIEW

**Comment by ehmicky** [comment]
Note: an alternative would be not to use `shell: true`. This means users won't be able to use shell features like `$variable` expansion or globbing, but it also means no escaping issues. It also is safer. See [more information](https://github.com/sindresorhus/execa#shell).

**Comment by RaeesBhatti** [comment]
Thanks for the tip @ehmicky. I attempted that yesterday but could make the addons tests work on Windows. Found out the problem today, we were trying to invoke a JS file without shell, which isn't supported. Updated now. Please take another look!

File: src/commands/dev/index.js [hunk 1]
```diff
@@ -342,14 +342,13 @@ async function startDevServer(settings, log) {
...
   log(`${NETLIFYDEVLOG} Starting Netlify Dev with ${settings.type}`)
   const args = settings.command === 'npm' ? ['run', ...settings.args] : settings.args
-  const ps = child_process.spawn(await which(settings.command), args, {
+  const commandBin = await which(settings.command)
+  const ps = child_process.spawn(commandBin, args, {
     env: { ...settings.env, FORCE_COLOR: 'true' },
-    stdio: settings.stdio || 'inherit',
-    detached: true,
-    shell: true,
+    stdio: 'pipe',
```
Comment (ehmicky) [inline src/commands/dev/index.js:348]: Is it possible for `settings.command` to ask for interactive input from the user?
If so, `stdio: 'pipe'` would not work.

Comment (RaeesBhatti) [reply src/commands/dev/index.js:348]: How about we do `process.stdin.pipe(ps.stdin)`?

Comment (RaeesBhatti) [reply src/commands/dev/index.js:348]: Implemented here: https://github.com/netlify/cli/pull/823/files

**Pull request merged**